packages/gatsby-plugin-mdx/src/get-source-plugins-as-remark-plugins.ts

```diff
@@ -32,30 +32,29 @@ export async function getSourcePluginsAsRemarkPlugins({
 
   const userPlugins = userPluginsFiltered.map(plugin => {
     const requiredPlugin = plugin.module
-    const wrappedGatsbyPlugin: Pluggable<Array<any>> =
-      function wrappedGatsbyPlugin() {
-        // eslint-disable-next-line @babel/no-invalid-this
-        const mdxNode = getNode(this.data(`mdxNodeId`) as string)
+    const wrappedGatsbyPlugin: Pluggable<any> = function wrappedGatsbyPlugin() {
+      // eslint-disable-next-line @babel/no-invalid-this
+      const mdxNode = getNode(this.data(`mdxNodeId`) as string)
 
-        return async function transformer(markdownAST): Promise<any> {
-          // Execute gatsby-remark-* plugin
-          await requiredPlugin(
-            {
-              markdownAST,
-              markdownNode: mdxNode,
-              getNode,
-              getNodesByType,
-              get files() {
-                return getNodesByType(`File`)
-              },
-              pathPrefix,
-              reporter,
-              cache,
+      return async function transformer(markdownAST): Promise<any> {
+        // Execute gatsby-remark-* plugin
+        await requiredPlugin(
+          {
+            markdownAST,
+            markdownNode: mdxNode,
+            getNode,
+            getNodesByType,
+            get files() {
+              return getNodesByType(`File`)
             },
-            plugin.options || {}
-          )
-        }
+            pathPrefix,
+            reporter,
+            cache,
+          },
+          plugin.pluginOptions || {}
```

Comment:
https://github.com/gatsbyjs/gatsby/pull/35650/files#diff-7304ffd26b36144767d37139257bf9707027d0fd8d8db379cfdae24700e24fe6L63

Shame on me. Thanks for finding and fixing!